tests/flytekit/unit/cli/pyflyte/test_run.py

```diff
@@ -429,6 +484,125 @@ def check_image(*args, **kwargs):
     run_command(mock_click_ctx, tk)()
 
 
+def jsons():
+    for x in [
+        {
+            "custom_id": "request-1",
+            "method": "POST",
+            "url": "/v1/chat/completions",
+            "body": {
+                "model": "gpt-3.5-turbo",
+                "messages": [
+                    {"role": "system", "content": "You are a helpful assistant."},
+                    {"role": "user", "content": "What is 2+2?"},
+                ],
+            },
+        },
+    ]:
+        yield x
+
+
+@mock.patch("flytekit.configuration.default_images.DefaultImages.default_image")
+def test_pyflyte_run_with_iterator_json_type(
+    mock_image, mock_image_spec_builder, caplog
```

Comment:
`caplog` ❤️ 

A lot of the tests in this test suite should use `caplog` instead. We can tackle this in a separate PR.